waku/waku_archive/driver/postgres_driver/postgres_driver.nim

```diff
@@ -240,6 +295,114 @@ method getMessages*(s: PostgresDriver,
+  ## prepared statements.
+  ##
+  ## contentTopic - string with list of conten topics. e.g: "'ctopic1','ctopic2','ctopic3'"
+
+  var rows: seq[(PubsubTopic, WakuMessage, seq[byte], Timestamp)]
+  proc rowCallback(pqResult: ptr PGresult) =
+    rowCallbackImpl(pqResult, rows)
+
+  let startTimeStr = $startTime
+  let endTimeStr = $endTime
+  let limit = $maxPageSize
+
+  if cursor.isSome():
+    var stmtName = if ascOrder: SelectWithCursorAscStmtName else: SelectWithCursorDescStmtName
+    var stmtDef = if ascOrder: SelectWithCursorAscStmtDef else: SelectWithCursorDescStmtDef
+
+    let digest = toHex(cursor.get().digest.data)
+    let storeTime = $cursor.get().storeTime
+
+    (await s.readConnPool.runStmt(
+                                  stmtName,
+                                  stmtDef,
+                                 @[contentTopic,
+                                   pubsubTopic,
+                                   storeTime,
+                                   digest,
+                                   startTimeStr,
+                                   endTimeStr,
+                                   limit],
+                                 @[int32(contentTopic.len),
+                                   int32(pubsubTopic.len),
+                                   int32(storeTime.len),
+                                   int32(digest.len),
+                                   int32(startTimeStr.len),
+                                   int32(endTimeStr.len),
+                                   int32(limit.len)],
+                                 @[int32(0), int32(0), int32(0), int32(0),
+                                   int32(0), int32(0), int32(0)],
+                                  rowCallback)
+    ).isOkOr:
+      return err("failed to run query with cursor: " & $error)
+
+  else:
+    var stmtName = if ascOrder: SelectNoCursorAscStmtName else: SelectNoCursorDescStmtName
+    var stmtDef = if ascOrder: SelectNoCursorAscStmtDef else: SelectNoCursorDescStmtDef
+    (await s.readConnPool.runStmt(stmtName,
+                                  stmtDef,
+                                 @[contentTopic,
+                                   pubsubTopic,
+                                   startTimeStr,
+                                   endTimeStr,
+                                   limit],
+                                 @[int32(contentTopic.len),
+                                   int32(pubsubTopic.len),
+                                   int32(startTimeStr.len),
+                                   int32(endTimeStr.len),
+                                   int32(limit.len)],
+                                 @[int32(0), int32(0), int32(0), int32(0), int32(0)],
+                                  rowCallback)
+    ).isOkOr:
+      return err("failed to run query without cursor: " & $error)
+
+  return ok(rows)
+
+method getMessages*(s: PostgresDriver,
```

Comment:
How come PostgresDriver reference named as "s"? :-)

Reply:
Thanks for the comments! I'll review that in upcoming refactoring PRs :)